Sources/Compiler/AST/Expr/SequenceExpr.swift

```diff
@@ -1,79 +1,50 @@
 /// A sequence of binary operations.
-public enum SequenceExpr: Expr {
+public struct SequenceExpr: Expr {
 
-  public typealias UnfoldedTail = [TailElement]
+  public static let kind = NodeKind.sequenceExpr
 
   /// The operator and right operand in an unfolded sequence of infix expressions.
   public struct TailElement: Codable {
 
-    /// The name of the operator.
-    public var operatorName: SourceRepresentable<Identifier>
+    /// The operator.
+    public var `operator`: NodeID<NameExpr>
 
     /// The right operand.
     public var operand: AnyExprID
 
-    public init(operatorName: SourceRepresentable<Identifier>, operand: AnyExprID) {
-      self.operatorName = operatorName
+    public init(operator: NodeID<NameExpr>, operand: AnyExprID) {
+      self.operator = `operator`
       self.operand = operand
     }
 
   }
 
-  /// A tree encoding sub-expression ordering of an unfolded `SequenceExpr`.
-  internal indirect enum Tree {
-
-    typealias Operator = (name: SourceRepresentable<Identifier>, precedence: PrecedenceGroup?)
+  /// The first operand of the expression.
+  public let head: AnyExprID
 
-    case node(operator: Operator, left: Tree, right: Tree)
+  /// A sequence of operator/operand pairs.
+  public let tail: [TailElement]
 
-    case leaf(AnyExprID)
-
-    /// Appends the RHS of an operation to `self`.
-    mutating func append(operator operator_: Operator, rhs: AnyExprID) {
-      switch self {
-      case .node(let lhsOperator, let lhsLeft, var lhsRight):
-        if let l = lhsOperator.precedence {
-          if let r = operator_.precedence {
-            // Both operators are in groups.
-            if (l < r) || (l == r && l.associativity == .left) {
-              self = .node(operator: operator_, left: self, right: .leaf(rhs))
-              return
-            }
+  /// Creates an instance with the given properties.
+  public init(head: AnyExprID, tail: [TailElement]) {
+    self.head = head
+    self.tail = tail
+  }
 
-            if (l > r) || (l == r && l.associativity == .right) {
-              lhsRight.append(operator: operator_, rhs: rhs)
-              self = .node(operator: lhsOperator, left: lhsLeft, right: lhsRight)
-              return
-            }
-          } else {
-            // Right operator is not in a group. Assume lowest precedence and left associativity.
-            self = .node(operator: operator_, left: self, right: .leaf(rhs))
-            return
-          }
-        } else if operator_.precedence != nil {
-          // Only right operator is in a group. Assume higher precedence.
-          lhsRight.append(operator: operator_, rhs: rhs)
-          self = .node(operator: lhsOperator, left: lhsLeft, right: lhsRight)
-        } else {
-          // Neither operator is in a group. Assume left associativity.
-          self = .node(operator: operator_, left: self, right: .leaf(rhs))
-        }
+  public func isWellFormed(in ast: AST) -> SuccessOrDiagnostics {
```

Comment:
```suggestion
  public func wellFormedness(in ast: AST) -> SuccessOrDiagnostics {
```
Anything but an `is` prefix for a non-`Bool` returning function.